main.tf

```diff
@@ -333,38 +392,38 @@ module "deep" {
 resource "spacelift_stack" "default" {
   for_each = local.stacks
 
-  administrative                   = coalesce(try(local.stack_configs[each.key].administrative, null), var.administrative)
-  additional_project_globs         = try(local.stack_configs[each.key].additional_project_globs, var.additional_project_globs)
-  after_apply                      = compact(concat(try(local.stack_configs[each.key].after_apply, []), var.after_apply))
-  after_destroy                    = compact(concat(try(local.stack_configs[each.key].after_destroy, []), var.after_destroy))
-  after_init                       = compact(concat(try(local.stack_configs[each.key].after_init, []), var.after_init))
-  after_perform                    = compact(concat(try(local.stack_configs[each.key].after_perform, []), var.after_perform))
-  after_plan                       = compact(concat(try(local.stack_configs[each.key].after_plan, []), var.after_plan))
-  after_run                        = compact(concat(try(local.stack_configs[each.key].after_run, []), var.after_run))
-  autodeploy                       = coalesce(try(local.stack_configs[each.key].autodeploy, null), var.autodeploy)
-  autoretry                        = try(local.stack_configs[each.key].autoretry, var.autoretry)
-  before_apply                     = compact(coalesce(try(local.stack_configs[each.key].before_apply, []), var.before_apply))
-  before_destroy                   = compact(coalesce(try(local.stack_configs[each.key].before_destroy, []), var.before_destroy))
+  administrative                   = local.stack_property_resolver[each.key].administrative
+  additional_project_globs         = local.stack_property_resolver[each.key].additional_project_globs
+  after_apply                      = local.stack_array_merger[each.key].after_apply
+  after_destroy                    = local.stack_array_merger[each.key].after_destroy
+  after_init                       = local.stack_array_merger[each.key].after_init
+  after_perform                    = local.stack_array_merger[each.key].after_perform
+  after_plan                       = local.stack_array_merger[each.key].after_plan
+  after_run                        = local.stack_array_merger[each.key].after_run
+  autodeploy                       = local.stack_property_resolver[each.key].autodeploy
+  autoretry                        = local.stack_property_resolver[each.key].autoretry
+  before_apply                     = local.stack_array_merger[each.key].before_apply
+  before_destroy                   = local.stack_array_merger[each.key].before_destroy
   before_init                      = local.before_init[each.key]
```

Comment:
@gberenice I believe we don't merge anything here because `before_init` is merged above. We should call that out in a comment. It looks odd.